packages/labs/observers/src/intersection_controller.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license
+ * Copyright 2017 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+import {ReactiveControllerHost} from '@lit/reactive-element/reactive-controller.js';
+
+export type IntersectionValueCallback = (
+  ...args: Parameters<IntersectionObserverCallback>
+) => unknown;
+
+export interface IntersectionControllerConfig {
+  config?: IntersectionObserverInit;
+  target?: Element;
+  callback?: IntersectionValueCallback;
+  skipInitial?: boolean;
+}
+
+export class IntersectionController {
+  private _host: ReactiveControllerHost;
+  private _target?: Element | null;
+  private _observer: IntersectionObserver;
+  private _skipInitial = false;
+  private _unobservedUpdate = false;
+  value?: unknown;
+  callback: IntersectionValueCallback = () => true;
+  constructor(
+    host: ReactiveControllerHost,
+    {target, config, callback, skipInitial}: IntersectionControllerConfig
+  ) {
+    this._host = host;
+    this._target = target;
+    if (skipInitial !== undefined) {
```

Comment:
style super-nit: I find this a little more readable: `this.skipInitial = skipInitial ?? false;`. ymmv

Reply:
Done.